CorsixTH/Lua/dialogs/fullscreen/town_map.lua

```diff
@@ -303,7 +306,10 @@ function UITownMap:draw(canvas, x, y)
       local owner_num = map.th:getPlotOwner(self.hover_plot)
       if owner_num == 0 then
         owner = _S.town_map.for_sale
-        price = "$" .. map:getParcelPrice(self.hover_plot)
+	price = "$" .. map:getParcelPrice(self.hover_plot)
+	if app.config.free_build_mode then  -- Everything is 0$ in free build mode
+	  price = "$0"
+	end
```

Comment:
As a general rule, try to avoid useless computations. A computer is fast but not infinitely fast, so use its time wisely.

Here you first construct a string with the actual price, which you then completely discard in case the free-build mode is active.

The better tactic is first to split on the case and then do the computation. An if/else or an and/or would work here.

If splitting is not easily possible and you want  something like you have here, first perform the cheaper case, and override that result with the more expensive case if necessary. In that way, if the override is needed you're throwing away less computation effort.

Reply:
Variant 1
```
price = "0$"
if not app.config.free_build_mode then
  price = "$" .. map:getParcelPrice(self.hover_plot) 
end
```
Variant 2
```
if app.config.free_build_mode then
  price = "0$"
else
  price = "$" .. map:getParcelPrice(self.hover_plot) 
end
```

Which variant is better?

Reply:
The 2nd variant has the advantage that it very explicit expresses the fact there are 2 independent cases.  
The 1st variant makes that more difficult, as a reader to have to verify that the word `price` above the `if` is the same as the word `price` within the `if`.

Obviously, here it's not very difficult, but image assigning to `something[self:computeIndex(123, 45, 65)][28].field.subfield` instead. Then comparing all the numbers and words is much more difficult to do manually. A typo is easily over-looked as humans are sloppy readers.

Reply:
In such a case the and/or approach is better, since you don't have to write the long assigned name twice. Alternatively, use a simple variable like `price` :p  and afterwards copy the value to the long name, ie

``` lua
local price = ....
something[self:computeIndex(123, 45, 65)][28].field.subfield = price
```
One additional line of code, but it avoids writing the assigned field twice.